cli/shared/src/main/scala/com/lightbend/rp/reactivecli/docker/DockerCredentials.scala

```diff
@@ -17,20 +17,89 @@
 package com.lightbend.rp.reactivecli.docker
 
 import com.lightbend.rp.reactivecli.files._
+import com.lightbend.rp.reactivecli.process._
 import scala.collection.immutable.Seq
+import scala.concurrent.{ Future, Await }
+import scala.concurrent.duration._
+import argonaut._
+import Argonaut._
 
-case class DockerCredentials(registry: String, username: String, password: String)
+case class DockerCredentials(registry: String, username: String, password: String, token: String)
 
 /**
- * Parses a file for credentials
+ * Finds docker credentials in multiple places:
+ * ~/.lightbend/docker.credentials
+ * ~/.docker/config.json
+ * OS-specific storage (OS X Keychan, Windows credential store, etc.)
  */
 object DockerCredentials {
   private val Registry = "registry"
   private val Username = "username"
   private val Password = "password"
 
-  def parse(path: String): Seq[DockerCredentials] =
-    decode(readFile(path))
+  def get(credsFilePath: Option[String], configFilePath: Option[String]): Seq[DockerCredentials] = {
+    // Credential priorities:
+    // 1. Lightbend credential file
+    // 2. Docker credential helpers
+    // 3. Docker config file
+    val fromCreds = credsFilePath.map(parseCredsFile).getOrElse(Seq.empty)
+    val fromHelpers = dockercred.getCredentials
+    val fromConfig = configFilePath.map(parseDockerConfig).getOrElse(Seq.empty)
+
+    def credsToMap(creds: Seq[DockerCredentials]): Map[String, DockerCredentials] = {
+      creds.map(c => c.registry -> c).toMap
+    }
+
+    // Combines two maps, keeps first one when keys overlap
+    def combine(first: Map[String, DockerCredentials], second: Map[String, DockerCredentials]) = {
```

Comment:
Can this method be removed and instead inlined as `second ++ first`, or does it have different semantics?

Reply:
Oh, I wasn't aware ++ does exactly what I need. Nice!